Custom inputs for updating RPC URLs

hkirat opened the pull request:
Closes https://github.com/coral-xyz/backpack/issues/941 

Awaiting review from design. 

## Screenshots - 

### Solana
<img width="371" alt="Screenshot 2022-10-06 at 2 18 12 AM" src="https://user-images.githubusercontent.com/8079861/194164461-f63e85a9-1897-4e08-82c8-0f3e9acf4369.png">
<img width="370" alt="Screenshot 2022-10-06 at 2 37 08 AM" src="https://user-images.githubusercontent.com/8079861/194164479-2bff4332-1403-4946-8cba-f6cd051f336f.png">
<img width="372" alt="Screenshot 2022-10-06 at 2 37 15 AM" src="https://user-images.githubusercontent.com/8079861/194164485-d8571664-7a42-42ae-b294-f1ab6fd3ed3b.png">
<img width="374" alt="Screenshot 2022-10-06 at 2 37 25 AM" src="https://user-images.githubusercontent.com/8079861/194164501-5a641c78-6687-4298-add9-1619a00cef78.png">


### Eth
<img width="371" alt="Screenshot 2022-10-06 at 2 37 37 AM" src="https://user-images.githubusercontent.com/8079861/194164547-79d2f793-53ba-484a-85b3-81f21bb0bcef.png">

<img width="374" alt="Screenshot 2022-10-06 at 2 37 54 AM" src="https://user-images.githubusercontent.com/8079861/194164538-45879aee-605f-4e4f-9dd5-100f1339eeaf.png">
<img width="372" alt="Screenshot 2022-10-06 at 2 37 43 AM" src="https://user-images.githubusercontent.com/8079861/194164542-10e3d363-79fc-4eb1-be29-b10188798335.png">

tomlinton commented:
This might be a separate PR, but it'd be nice if we the Custom option allowed you to persist these into a list that showed up below (effectively allowing multiple custom RPC settings with the ability to quickly switch between them). This is something that is super useful as a developer, but might not be useful to anyone else.

hkirat commented:
@tomlinton I think that makes a lot of sense. 
Just so I understand this correctly, you mean we - 
1. Store these custom RPCs locally
2. If a user has more than 0 of these stored locally, we show these as a list below the currently available 4 options as a `Recently used` set of RPCs?

tomlinton commented:
Yea, and `Custom` is an `Add Custom` button, and you can delete/edit things in the custom list. We can punt this for now though, as what you've got should cover most use cases.

0xInq commented on packages/app-extension/src/components/Unlocked/Settings/Preferences/Ethereum/CustomRpcUrl.tsx:
```diff
@@ -0,0 +1,136 @@
+                padding: "10px",
+              }}
+            >
+              <Typography style={{ width: "80px" }}>Chian ID</Typography>
```
just a typo, this should be `Chain ID`.

hkirat commented:
cc @trucy @aarongillett for design review

trucy commented:
Link to Figma with input states in light and dark mode: https://www.figma.com/file/HNF2HJZYq6dUzCWxSmH2Oe/Backpack-_-Questlog?node-id=12036%3A44072

Related issue that would be fixed with this work is https://github.com/coral-xyz/backpack/issues/880

hkirat commented:
Updated as per designs to close #880 along with this one. 
@trucy you can find the new extension attached for review
[dev.zip](https://github.com/coral-xyz/backpack/files/9795790/dev.zip)

Have made changes across a lot of files to use the new `input styles`

tomlinton commented:
Is there a need to add a separate TextInput component here, was it for styling reasons? I think in general we want to move all our components to those from `react-xnft-renderer` so that the design is consistent across the wallet and xNFTs.

hkirat commented:
So after https://github.com/coral-xyz/backpack/pull/1023 gets merged in the `react-xnft-renderer` gets purged and all components move to `react-xnft-dom-renderer`.
The `react-xnft-dom-renderer` would probably be moved to use `tailwind`, and maybe even support the latest version of react so we might not be able to share components across the core app and the xnft renders. 
I think this is by design since we want to move fast on the `xnft renderer` and provide a lot of flexibility there. 

So as you'll see in https://github.com/coral-xyz/backpack/pull/1023 I separated out the `Components.tsx` from `react-xnft-renderer` to `app-extensions` because of the same reason (They're basically duplicated right now, but the former would probably diverge out to use tailwind)

tomlinton commented:
Ahh great, makes sense!

hkirat commented:
cc @trucy for review

trucy commented:
Hey @hkirat thanks for the update! I downloaded the attachment and ran it but I'm not sure if I'm seeing the changes across inputs. I still see the focus/active state border showing up as Base/900 (#030A19), when it should be (#787C89) in a few places
<img width="386" alt="Screen Shot 2022-10-25 at 3 27 18 PM" src="https://user-images.githubusercontent.com/321395/197876741-44a6ae1e-b328-49d1-9c43-9931ff6dbf74.png">
<img width="379" alt="Screen Shot 2022-10-25 at 3 26 03 PM" src="https://user-images.githubusercontent.com/321395/197876743-6b1a5548-c14d-42b1-91fb-a3c0aa789e54.png">

I also noticed the top two corners of the grouped input for Change RPC Connection look square for both Ethereum and Solana
<img width="376" alt="Screen Shot 2022-10-25 at 3 29 46 PM" src="https://user-images.githubusercontent.com/321395/197876913-696e69ec-9f9b-4816-ae6f-c9d1d85b29c5.png">
<img width="375" alt="Screen Shot 2022-10-25 at 3 29 28 PM" src="https://user-images.githubusercontent.com/321395/197876915-a41989f7-f981-4d5d-93b9-7be50b52b7e1.png">

hkirat commented:
Great 👁️ @trucy, added input styles in the places you mentioned @trucy . Unfortunately there was no centralised place for all these inputs until now so these got missed (and there might be other places where it still got missed, although I've tried to manually go through all paths to ensure there aren't). The good thing is that now there's a centralised place so any future style changes should be propagated to all the places.

<img width="867" alt="Screenshot 2022-10-30 at 3 14 16 AM" src="https://user-images.githubusercontent.com/8079861/198853964-c853806a-207e-450e-a466-421fa9aea51e.png">


<img width="376" alt="Screenshot 2022-10-30 at 3 29 02 AM" src="https://user-images.githubusercontent.com/8079861/198853948-b686b3fd-caa6-4c4e-9413-4ee80e38bf86.png">

Fixed the borders you mentioned above as well 
<img width="390" alt="Screenshot 2022-10-30 at 3 28 37 AM" src="https://user-images.githubusercontent.com/8079861/198853896-1b7e6bdb-d785-4765-a21b-1baa283ed547.png">
<img width="386" alt="Screenshot 2022-10-30 at 3 28 28 AM" src="https://user-images.githubusercontent.com/8079861/198853943-bcd3c41f-9504-4ce4-9d22-e1cb5b99414c.png">

Might merge this in soon since it's becoming a little big and goes out of date with master very quickly, but will follow up with any places the styles might've gotten missed.
